integration/spark/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -228,8 +230,15 @@ class CarbonFileMetastore extends CarbonMetaStore {
             c.getClass.getName.equals("org.apache.spark.sql.catalyst.catalog.HiveTableRelation") ||
             c.getClass.getName.equals(
               "org.apache.spark.sql.catalyst.catalog.UnresolvedCatalogRelation")) =>
-        val catalogTable =
+        var catalogTable =
           CarbonReflectionUtils.getFieldOfCatalogTable("tableMeta", c).asInstanceOf[CatalogTable]
+        // remove spatial column from schema
```

Comment:
same as above, we can register this column as it is visible column

Reply:
Here, catalogTable will have spatial column in schema which is used to build carbon table.  As spatial column is not supposed to be present in user-defined columns, removing it here. Later from tableproperties the column will be added in carbonTable.

Reply:
would have added this as comment !